python/pyarrow/parquet/core.py

```diff
@@ -3430,10 +3430,14 @@ def write_metadata(schema, where, metadata_collector=None, **kwargs):
     if metadata_collector is not None:
         # ParquetWriter doesn't expose the metadata until it's written. Write
         # it and read it again.
-        metadata = read_metadata(where)
+        metadata = read_metadata(where, filesystem=kwargs.get("filesystem"))
```

Comment:
I think we can still explicitly add a filesystem keyword to the `write_metadata` signature instead through kwargs; that makes it more discoverable for the user (and then also document it in the docstring), and is consistent with other functions in this module that support this keyword.

Reply:
Oi, I misinterpreted this one. Updated now. :) https://github.com/apache/arrow/pull/14574/commits/e120e9ada309d83df674be22e2106e8e056cab44